### PR TITLE
Handle migration of `containerCap = 0`

### PR DESCRIPTION
**Pull request by Vlatombe**
Amends #866. Serialized 0 means unlimited.

@schottsfired @twasyl @kerogers-cloudbees

### CODE REVIEW

**Review by kerogers-cloudbees — APPROVED** [review]
I feel like I would not have known how to write the config.xml for the test to use.

**Comment by jglick** [comment]
> I feel like I would not have known how to write the config.xml for the test to use.

You do not write it, you copy it from an actual `$JENKINS_HOME`.

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloudTest.java [hunk 1]
```diff
@@ -372,6 +372,13 @@ public void emptyKubernetesCloudReadResolve() {
         assertEquals(KubernetesCloud.DEFAULT_WAIT_FOR_POD_SEC, cloud.getWaitForPodSec());
     }
 
+    @Test
+    @LocalData
+    public void readResolveContainerCapZero() {
+        KubernetesCloud cloud = j.jenkins.clouds.get(KubernetesCloud.class);
+        assertTrue(cloud.getRemainingGlobalSlots(Collections.emptyList(), 1) > 0);
```
Comment (jglick) [inline src/test/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloudTest.java:379]: There is a `Matcher` for this, FWIW.

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloudTest.java [hunk 2]
```diff
@@ -372,6 +372,13 @@ public void emptyKubernetesCloudReadResolve() {
         assertEquals(KubernetesCloud.DEFAULT_WAIT_FOR_POD_SEC, cloud.getWaitForPodSec());
     }
 
+    @Test
+    @LocalData
+    public void readResolveContainerCapZero() {
+        KubernetesCloud cloud = j.jenkins.clouds.get(KubernetesCloud.class);
```
Comment (jglick) [inline src/test/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloudTest.java:378]: Maybe start by checking `getContainerCap`?

**Review by jglick — APPROVED** [review]

**Comment by jglick** [comment]
@schottsfired please link to historical PRs in this area for archaeological purposes

**Review by jglick — APPROVED** [review]

**Comment by Vlatombe** [comment]
CI infra issue, tests were fine.

**Comment by schottsfired** [comment]
Hm, so the `readResolve` impl was necessary. Thanks a bunch @Vlatombe 👍

**Pull request merged**